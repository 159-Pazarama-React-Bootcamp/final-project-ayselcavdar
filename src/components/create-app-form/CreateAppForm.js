import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import styles from './CreateAppForm.module.css';
import FormInput from '../form-input/FormInput';
import { useWindowSize } from '../../hooks/useWindowSize';
import { useState } from 'react';
import { imgHandler, removeImg } from '../../helper/imgUpload';
import { useNavigate } from 'react-router-dom';
import Button from '../button/Button';
import { useDispatch } from 'react-redux';
import { createApplication } from '../../redux/actions/crudActions';
import { useSelector } from 'react-redux';

const CreateAppForm = () => {
  const [profileImg, setProfileImg] = useState(null);
  const [width] = useWindowSize();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { currUser } = useSelector(state => state.user)
  const initialValues = {
    firstName: '',
    lastName: '',
    age: '',
    tcNum: '',
    reason: '',
    address: '',
    photo: '',
  };

  const validate = Yup.object({
    firstName: Yup.string()
      .trim()
      .min(2, 'Cok kisa')
      .max(50, '50 karakter veya daha az olmalı')
      .required('Zorunlu alan'),
    lastName: Yup.string()
      .trim()
      .min(2, 'Cok kisa')
      .max(50, '50 karakter veya daha az olmalı')
      .required('Zorunlu alan'),
    age: Yup.number()
      .min(18, '18 veya daha üzeri olmalı')
      .max(105, '105 veya daha altı olmalı')
      .required('Zorunlu alan'),
    tcNum: Yup.string()
      .trim()
      .matches(
        /^[1-9][0-9]*$/,
        'Tc kimlik numarası yalnızca rakamlardan olusmalıdır ve sıfır ile baslayamaz.'
      )
      .min(11, 'Tc kimlik numarası 11 haneli olmalıdır')
      .max(11, 'Tc kimlik numarası 11 haneli olmalıdır')
      .required('Zorunlu alan'),
    reason: Yup.string()
      .trim()
      .min(20, '20 karakter veya daha üzeri olmalı')
      .max(255, '255 karater veya daha altı olmalı')
      .required('Zorunlu alan'),
    address: Yup.string()
      .trim()
      .min(20, '20 karakter veya daha üzeri olmalı')
      .max(255, '255 karater veya daha altı olmalı')
      .required('Zorunlu alan'),
  });

  const handleSubmit = (values) => {
    values['photo'] = profileImg;
    dispatch(createApplication(values));
    navigate('/basvurular');
  };

  return (
    <div
      className={styles['application-container']}
      style={{
        justifyContent: width < 1024 && 'center',
        height: !profileImg && '100vh',
      }}
    >
      <div>
        <Formik
          initialValues={initialValues}
          validationSchema={validate}
          onSubmit={(values, { resetForm }) => {
            handleSubmit(values);
            resetForm({});
          }}
        >
          {() => (
            <div>
              <h1
                style={{
                  textAlign: width < 1024 && 'center',
                  color: 'rgb(107, 99, 255)',
                }}
              >
                Basvuru Olustur
              </h1>
              <Form>
                <FormInput label="Ad" name="firstName" type="text" />
                <FormInput label="Soyad" name="lastName" type="text" />
                <FormInput label="Yaş" name="age" type="number" />
                <FormInput label="TC" name="tcNum" type="text" />
                <FormInput
                  label="Basvuru Nedeni"
                  name="reason"
                  type="textarea"
                />
                <FormInput label="Adress" name="address" type="textarea" />
                <div className={styles['main-file--container']}>
                  <div className={styles['file-input--container']}>
                    <FormInput
                      label="Fotograf Yükle"
                      name="photo"
                      type="file"
                      accept="image/*"
                      onChange={(e) => imgHandler(e, setProfileImg)}
                      className={styles['file-input']}
                    />
                  </div>
                  {profileImg && (
                    <div className={styles['uploaded-img--container']}>
                      <div
                        className={styles['remove-image']}
                        onClick={() => removeImg(setProfileImg)}
                      >
                        &#x2715;
                      </div>
                      <img
                        src={profileImg}
                        alt="uploaded_image"
                        width={'160px'}
                        height={'160px'}
                      />
                    </div>
                  )}
                </div>
                <div className={styles['btn-container']}>
                  <Button type="submit" content={'Gönder'} />
                  <Button
                    type="reset"
                    onClick={() => navigate(currUser ? 'admin/basvuru-listesi' : '/basvurular')}
                    content={currUser ? 'Basvurular' : 'Basvurularım'}
                  />
                </div>
              </Form>
            </div>
          )}
        </Formik>
      </div>
      <div>
        <img
          src={require('../../assets/images/form1.png')}
          style={{ display: width < 1024 && 'none', borderRadius: '1rem' }}
          width={width / 2}
        />
      </div>
    </div>
  );
};

export default CreateAppForm;
