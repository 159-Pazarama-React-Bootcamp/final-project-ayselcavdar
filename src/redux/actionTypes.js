const LOGIN_START = 'LOGIN_START';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_FAIL = 'LOGIN_FAIL';

const LOGOUT_START = 'LOGOUT_START';
const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
const LOGOUT_FAIL = 'LOGOUT_FAIL';

const SET_USER = 'SET_USER';

const GET_APPLICATIONS_START = 'GET_APPLICATIONS_START';
const GET_APPLICATIONS_SUCCESS = 'GET_APPLICATIONS_SUCCESS';
const GET_APPLICATIONS_FAIL = 'GET_APPLICATIONS_FAIL';

const DELETE_APPLICATION_START = 'DELETE_APPLICATION_START';
const DELETE_APPLICATION_FAIL = 'DELETE_APPLICATION_FAIL';
const DELETE_APPLICATION_SUCCESS = 'DELETE_APPLICATION_SUCCESS';

const CREATE_APPLICATION_START = 'CREATE_APPLICATION_START';
const CREATE_APPLICATION_SUCCESS = 'CREATE_APPLICATION_SUCCESS';
const CREATE_APPLICATION_FAIL = 'CREATE_APPLICATION_FAIL';

const EDIT_APPLICATION_START = 'EDIT_APPLICATION_START';
const EDIT_APPLICATION_FAIL = 'EDIT_APPLICATION_FAIL';
const EDIT_APPLICATION_SUCCESS = 'EDIT_APPLICATION_SUCCESS';


export {
  LOGIN_START,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_START,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL,
  SET_USER,
  CREATE_APPLICATION_FAIL,
  CREATE_APPLICATION_START,
  CREATE_APPLICATION_SUCCESS,
  GET_APPLICATIONS_FAIL,
  GET_APPLICATIONS_START,
  GET_APPLICATIONS_SUCCESS,
  DELETE_APPLICATION_START,
  DELETE_APPLICATION_SUCCESS,
  DELETE_APPLICATION_FAIL,
  EDIT_APPLICATION_FAIL,
  EDIT_APPLICATION_START,
  EDIT_APPLICATION_SUCCESS,
};
