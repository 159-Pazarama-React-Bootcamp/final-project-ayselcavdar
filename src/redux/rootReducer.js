import {combineReducers} from 'redux';
import { authReducer } from './reducers/authReducer';
import applicationReducer from './reducers/applicationReducer';

const reducers = {
  user: authReducer,
  data: applicationReducer,
};

export const rootReducer = combineReducers(reducers);