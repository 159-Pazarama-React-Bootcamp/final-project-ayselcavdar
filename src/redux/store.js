import {createStore, applyMiddleware} from 'redux';
import {rootReducer} from './rootReducer';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middleware = [thunk];
// state değişikliklerini takibe yarayan anlık loglar sunan logger modulü,
// yalnızca development ortamı için uygulanmıştır
if(process.env.NODE_ENV === 'development'){
    middleware.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middleware));