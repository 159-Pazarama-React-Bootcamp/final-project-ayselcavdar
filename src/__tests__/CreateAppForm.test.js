/**
 * @jest-environment jsdom
 */
import render from 'react-test-renderer';
import CreateAppForm from '../components/create-app-form/CreateAppForm';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

describe('<AppDetails /> rendering', () => {
  test('AppDetails snapshots check', () => {
    const wrapper = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <CreateAppForm />
        </BrowserRouter>
      </Provider>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
