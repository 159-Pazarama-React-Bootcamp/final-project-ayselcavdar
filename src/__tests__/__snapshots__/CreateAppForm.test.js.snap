// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppDetails /> rendering AppDetails snapshots check 1`] = `
<div
  className="application-container"
  style={
    Object {
      "justifyContent": false,
    }
  }
>
  <div>
    <p
      className="already-have"
    >
      Mevcut başvurularımı görüntülemek istiyorum.
    </p>
    <div
      className="query-container"
    >
      <div
        className="form-container custom-input"
      >
        <label
          className="form-container--label"
          htmlFor="tc"
        >
          
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="tc"
          name="tc"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Tc kimlik numaranızı giriniz"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Görüntüle
      </button>
    </div>
    <hr />
    <h1
      style={
        Object {
          "color": "rgb(107, 99, 255)",
          "marginTop": "0px",
          "textAlign": false,
        }
      }
    >
      Basvuru Olustur
    </h1>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="firstName"
        >
          Ad
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="firstName"
          name="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="lastName"
        >
          Soyad
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="lastName"
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="age"
        >
          Yaş
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="age"
          name="age"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="tcNum"
        >
          TC
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="tcNum"
          name="tcNum"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="reason"
        >
          Basvuru Nedeni
        </label>
        <textarea
          className="form-control false "
          cols="50"
          id="reason"
          name="reason"
          onBlur={[Function]}
          onChange={[Function]}
          rows="2"
          type="textarea"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="address"
        >
          Adress
        </label>
        <textarea
          className="form-control false "
          cols="50"
          id="address"
          name="address"
          onBlur={[Function]}
          onChange={[Function]}
          rows="2"
          type="textarea"
          value=""
        />
      </div>
      <div
        className="main-file--container"
      >
        <div
          className="file-input--container"
        >
          <label
            style={
              Object {
                "marginBottom": "1rem",
              }
            }
          >
            Fotoğraf
          </label>
          <input
            accept="image/*"
            name="photo"
            onChange={[Function]}
            type="file"
          />
        </div>
        
      </div>
      <div
        className="btn-container"
      >
        <button
          className="btn"
          style={Object {}}
          type="submit"
        >
          Gönder
        </button>
      </div>
    </form>
  </div>
  <div>
    <img
      src="test-file-stub"
      style={
        Object {
          "borderRadius": "1rem",
          "display": false,
        }
      }
      width={512}
    />
  </div>
</div>
`;

exports[`AppDetails render check 1`] = `
<div
  className="application-container"
  style={
    Object {
      "justifyContent": false,
    }
  }
>
  <div>
    <p
      className="already-have"
    >
      Mevcut başvurularımı görüntülemek istiyorum.
    </p>
    <div
      className="query-container"
    >
      <div
        className="form-container custom-input"
      >
        <label
          className="form-container--label"
          htmlFor="tc"
        >
          
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="tc"
          name="tc"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Tc kimlik numaranızı giriniz"
          type="text"
          value=""
        />
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Görüntüle
      </button>
    </div>
    <hr />
    <h1
      style={
        Object {
          "color": "rgb(107, 99, 255)",
          "marginTop": "0px",
          "textAlign": false,
        }
      }
    >
      Basvuru Olustur
    </h1>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="firstName"
        >
          Ad
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="firstName"
          name="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="lastName"
        >
          Soyad
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="lastName"
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="age"
        >
          Yaş
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="age"
          name="age"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="tcNum"
        >
          TC
        </label>
        <input
          autoComplete="off"
          className="form-control false"
          id="tcNum"
          name="tcNum"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="reason"
        >
          Basvuru Nedeni
        </label>
        <textarea
          className="form-control false "
          cols="50"
          id="reason"
          name="reason"
          onBlur={[Function]}
          onChange={[Function]}
          rows="2"
          type="textarea"
          value=""
        />
      </div>
      <div
        className="form-container "
      >
        <label
          className="form-container--label"
          htmlFor="address"
        >
          Adress
        </label>
        <textarea
          className="form-control false "
          cols="50"
          id="address"
          name="address"
          onBlur={[Function]}
          onChange={[Function]}
          rows="2"
          type="textarea"
          value=""
        />
      </div>
      <div
        className="main-file--container"
      >
        <div
          className="file-input--container"
        >
          <label
            style={
              Object {
                "marginBottom": "1rem",
              }
            }
          >
            Fotoğraf
          </label>
          <input
            accept="image/*"
            name="photo"
            onChange={[Function]}
            type="file"
          />
        </div>
        
      </div>
      <div
        className="btn-container"
      >
        <button
          className="btn"
          style={Object {}}
          type="submit"
        >
          Gönder
        </button>
      </div>
    </form>
  </div>
  <div>
    <img
      src="test-file-stub"
      style={
        Object {
          "borderRadius": "1rem",
          "display": false,
        }
      }
      width={512}
    />
  </div>
</div>
`;
